### ISSUE TITLE
typevar bindings disappear in "indirect" proxy

### ISSUE THREAD

**Issue by GreyElaina** [microsoft/pyright#5409]
**Describe the bug**
```py
from __future__ import annotations

from typing import TypeVar, Callable, Protocol
from typing_extensions import ParamSpec

P = ParamSpec("P")
R = TypeVar("R", covariant=True)


class K(Protocol[P, R]):
    def collect(self, *args: P.args, **kwargs: P.kwargs) -> R:
        ...

def x(a: K[P, R], *args: P.args, **kwargs: P.kwargs) -> R:
    ...


T = TypeVar("T")

class X:
    @classmethod
    def collect(cls, n: type[T]):
        def receiver(entity: Callable[[T], int]):
            ...
        return receiver

reveal_type(x(X, float))  # Type of "x(X, float)" is "(entity: (T@collect) -> int) -> None"
```

**To Reproduce**
upper.

**Expected behavior**
we get `(entity: (float) -> int) -> None` or something makes sense.

**Code or Screenshots**
upper.

**VS Code extension or command-line**
pylance, latest

**Additional context**

**Comment by GreyElaina** [comment]
even we use `__call__` and `reveal_type(x(X(), int))`, still.

**Comment by erictraut** [comment]
Thanks for the bug report. I have a fix, and it will be included in the next release of pyright.

**Comment by erictraut** [comment]
This is included in pyright 1.1.317, which I just published. It will also be included in future releases of pylance.

**Comment by GreyElaina** [comment]
ohhhhhhhhh thanksssssssssssssssssssssssssssssssssss
